relayer/relays/execution/main.go

```diff
@@ -198,6 +195,41 @@ func (r *Relay) Start(ctx context.Context, eg *errgroup.Group) error {
 	}
 }
 
+func (r *Relay) writeToParachain(ctx context.Context, proof scale.ProofPayload, inboundMsg *parachain.Message) error {
+	inboundMsg.Proof.ExecutionProof = proof.HeaderPayload
+
+	log.WithFields(logrus.Fields{
+		"EventLog": inboundMsg.EventLog,
+		"Proof":    inboundMsg.Proof,
+	}).Debug("Generated message from Ethereum log")
+
+	// There is already a valid finalized header on-chain that can prove the message
+	if proof.FinalizedPayload == nil {
+		err := r.writer.WriteToParachainAndWatch(ctx, "EthereumInboundQueue.submit", inboundMsg)
+		if err != nil {
+			return fmt.Errorf("submit message to inbound queue: %w", err)
+		}
+
+		return nil
+	}
+
+	log.WithFields(logrus.Fields{
+		"finalized_slot": proof.FinalizedPayload.Payload.FinalizedHeader.Slot,
+		"finalized_root": proof.FinalizedPayload.FinalizedHeaderBlockRoot,
+		"message_slot":   proof.HeaderPayload.Header.Slot,
+	}).Debug("Batching finalized header update with message")
+
+	extrinsics := []string{"EthereumBeaconClient.submit", "EthereumInboundQueue.submit"}
+	payloads := []interface{}{proof.FinalizedPayload.Payload, inboundMsg}
+	// Batch the finalized header update with the inbound message
+	err := r.writer.BatchCall(ctx, extrinsics, payloads)
```

Comment:
Seems we use execution relayer to submit the beacon finality update. 

I would suggest not do like this, in recent change of beefy relayer we add a one-shot command to sync finality update.  Maybe we can follow the same pattern.

More context in https://github.com/Snowfork/snowbridge/pull/1217#issue-2326816048